feat: add circuit info on trackmania infobox league

IridiuMAG opened the pull request:
## Summary

Add circuit parameters to trackmania infobox league for new TMWT circuit

<!--
 Explain the **motivation** for making this change. What problems are you solving with this pull request? How are you improving the current situation?

 Please also consider the diff size. If you have changed more than 100 lines, chances are your pull request will be almost impossible to review. Are there any ways to
 split up your pull request further? Does the collection of changes semantically make sense?
-->

## How did you test this change?

dev
<!--
  Demonstrate the code is solid. Example: The exact pages you used to test this change, screenshots / videos if the pull request changes the user interface.
  How exactly did you verify that your PR solves the issue you wanted to solve?
  If you leave this empty, your PR will very likely be closed.

  Things to be particularly aware of:

  - Does this break LPDB on any of the wikis?
  - Are all needed page variables still set?
-->

hjpalpha commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -116,6 +122,10 @@ function CustomLeague:defineCustomPageVariables(args)
 	Variables.varDefine('date', self.data.endDate)
 	Variables.varDefine('sdate', self.data.startDate)
 	Variables.varDefine('edate', self.data.endDate)
+
+	Variables.varDefine('circuit', self.data.circuit)
+	Variables.varDefine('circuittier', self.data.circuittier)
+	Variables.varDefine('circuitabbr', self.data.circuitabbr)
```
where are those used? // are they needed?

IridiuMAG replied:
They are used in pages where we need to define the circuit. See https://liquipedia.net/trackmania/BIG_LAN_Race for an example

hjpalpha replied:
to me it looks like they are not used at all

https://liquipedia.net/trackmania/index.php?title=Special:Search&profile=all&search=insource%3A%22circuitabbr%22&fulltext=1

currently there are no consumers for these wiki variables

IridiuMAG replied:
Removed.

hjpalpha commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -140,10 +150,50 @@ function CustomLeague:addToLpdb(lpdbData, args)
 end
 
+---@param widgets Widget[]
+---@param circuitIndex string|number?
```
if this is optional you will get errors at lines 168-172

IridiuMAG replied:
Won't line 165 prevent this? I am setting it to null if its omitted from the call

hjpalpha replied:
ah, right, didn't see it on phone

hjpalpha reviewed (commented):
is there a reason for the `circuitIndex` stuff?
`_createCircuitInformation` doesn't get called with a set `circuitIndex` so it seems redundant to me

IridiuMAG commented:
> is there a reason for the `circuitIndex` stuff? `_createCircuitInformation` doesn't get called with a set `circuitIndex` so it seems redundant to me

I added it as a sort of placeholder in case we ever need to add a 2nd circuit to a tournament. Looking at it, I think it'll be quite easy to add this at a later point of time. I've removed it.

hjpalpha commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -140,10 +149,46 @@ function CustomLeague:addToLpdb(lpdbData, args)
+
+	return self:createSeriesDisplay({
+		displayManualIcons = true,
+		series = args['circuit'],
+	}, self.data['circuitIconDisplay'])
```
```suggestion
		series = args.circuit,
	}, self.data.circuitIconDisplay)
```

hjpalpha commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -140,10 +149,46 @@ function CustomLeague:addToLpdb(lpdbData, args)
+---@param widgets Widget[]
+function CustomLeague:_createCircuitInformation(widgets)
+	local args = self.args
+	local circuitArgs = {
+		tier = args['circuittier'],
+		region = args['region'],
+		points = args['points'],
+		next = args['circuit_next'],
+		previous = args['circuit_previous'],
+	}
+
+	Array.appendWith(widgets,
+		Cell{
+			name = 'Circuit',
+			content = {self:_createCircuitLink()}
+		},
+		Cell{name = 'Circuit Tier', content = {circuitArgs.tier}},
+		Cell{name = 'Tournament Region', content = {circuitArgs.region}},
+		Cell{name = 'Points', content = {circuitArgs.points}},
+		Chronology{content = {next = circuitArgs.next, previous = circuitArgs.previous}}
+	)
```
```suggestion
	Array.appendWith(widgets,
		Cell{
			name = 'Circuit',
			content = {self:_createCircuitLink()}
		},
		Cell{name = 'Circuit Tier', content = {args.circuittier}},
		Cell{name = 'Tournament Region', content = {args.region}},
		Cell{name = 'Points', content = {args.points}},
		Chronology{content = {next = args.circuit_next, previous = args.circuit_previous}}
	)
```

should do the same and is quite a bit shorter

hjpalpha commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -116,6 +122,9 @@ function CustomLeague:defineCustomPageVariables(args)
 	Variables.varDefine('date', self.data.endDate)
 	Variables.varDefine('sdate', self.data.startDate)
 	Variables.varDefine('edate', self.data.endDate)
+
+	Variables.varDefine('circuit', self.data.circuit)
+	Variables.varDefine('circuittier', self.data.circuittier)
```
```suggestion
```
don't see any usage on the wiki for the wiki variables
if there is planned usage for them let us know that
mind this is only about the wiki variables not about storage in lpdb
wiki variables are used to pass down the information on the same page the infobox is set and nowhere else

IridiuMAG replied:
Don't think it will be used either. Removed

hjpalpha reviewed (approved):
okay for me

IridiuMAG commented:
Hello, would appreciate if this was merged since we will already have the TMWT circuit started from this weekend. I would like to create the required pages without the use of dev infoboxes and modules

mbergen commented on components/infobox/wikis/trackmania/infobox_league_custom.lua:
```diff
@@ -140,10 +146,38 @@ function CustomLeague:addToLpdb(lpdbData, args)
+	return self:createSeriesDisplay({
+		displayManualIcons = true,
+		series = args.circuit,
+	}, self.data.circuitIconDisplay)
```
Where is the circuitIconDisplay coming from?

IridiuMAG replied:
I don't think it's really come from anywhere. It's the storage for the circuitIconDisplay param from the createSeriesDisplay call. It should work even without the circuitIconDisplay call

mbergen replied:
Then i'd  say remove it
```suggestion
	})
```
